ml-agents/mlagents/trainers/saver/tf_saver.py

```diff
@@ -0,0 +1,145 @@
+import os
+import shutil
+from typing import Optional
+from mlagents_envs.exception import UnityPolicyException
+from mlagents_envs.logging_util import get_logger
+from mlagents.tf_utils import tf
+from mlagents.trainers.saver.saver import BaseSaver
+from mlagents.trainers.tf.model_serialization import export_policy_model
+from mlagents.trainers.settings import TrainerSettings, SerializationSettings
+from mlagents.trainers.policy.tf_policy import TFPolicy
+from mlagents.trainers import __version__
+
+
+logger = get_logger(__name__)
+
+
+class TFSaver(BaseSaver):
+    """
+    Saver class for TensorFlow
+    """
+
+    def __init__(
+        self, trainer_settings: TrainerSettings, model_path: str, load: bool = False
+    ):
+        super().__init__()
+        self.model_path = model_path
+        self.initialize_path = trainer_settings.init_path
+        self._keep_checkpoints = trainer_settings.keep_checkpoints
+        self.load = load
+
+        # Currently only support saving one policy. This is the one to be saved.
+        self.policy: Optional[TFPolicy] = None
+
+    def register(self, module, init_or_load=True):
+        if isinstance(module, TFPolicy):
+            if self.policy is None:
+                self.policy = module
+                self.graph = self.policy.graph
```

Comment:
nit: initialize self.graph, self.sess, and self.tf_saver to None in `__init__()`

Reply:
done.